sed/core/workflow_recorder.py

```diff
@@ -0,0 +1,32 @@
+"""Module contains a call tracker and a dataclass for saving the method calls."""
+from dataclasses import dataclass
+from functools import update_wrapper
+from typing import List
+
+
+@dataclass
```

Comment:
Although I'm a big fan of the "dataclasses" module, to be consistent over the whole package,  either I would stick to the way you were generating special methods (very easy) or adopt dataclasses everywhere (too much work).

In any case, LGTM.

Reply:
Hmm could you elaborate on the special methods? I just thought this is a very good usecase for a dataclass. Perhaps, if you give examples of where else it could be used in our code, it could be implemented. I personally like the clarity they bring to the codebase.

Reply:
Special methods are, for example, all double under methods such as the ones used to initialize a class (`__init__(self, ...)`.

When I'm writing a class I tend to use the `dataclass` decorator because it automatically takes care of some of those special methods that are boring to write :)

As far as I can see you only use it here and, from my point of view, it breaks consistency with all the other classes already implemented throughout all the library.

Reply:
My thought was to use the dataclasses for classes that only hold data and the processing elsewhere. I especially see a good usecase with the Metadata as dataclasses can easily have entries like value and unit (using field). Of course, it's possible to do with dicts but it's not possible to add features to them. 
Perhaps I had a wrong idea about dataclasses and you can correct me.

Reply:
As I said, I tend to use them because it's a handy decorator. But it only does that (avoiding to write boilerplate code).

My point was that you tend to use dictionaries, lists and such in other classes, so personally, I would continue with such style also in here. I don't see any problems adding new entries to dictionaries or lists.

But, that's my personal view. I tend to be consistent with the code I write ;)